geo-data-server/src/test/java/com/powsybl/geodata/server/EmbeddedCassandraFactoryConfig.java

```diff
@@ -0,0 +1,30 @@
+package com.powsybl.geodata.server;
+
+import com.github.nosan.embedded.cassandra.EmbeddedCassandraFactory;
+import com.github.nosan.embedded.cassandra.api.CassandraFactory;
+import com.github.nosan.embedded.cassandra.api.Version;
+import com.github.nosan.embedded.cassandra.artifact.DefaultArtifact;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Configuration
+public class EmbeddedCassandraFactoryConfig {
+
+    @Bean
+    @Scope("singleton")
+    CassandraFactory embeddedCassandraFactory() throws UnknownHostException {
+        EmbeddedCassandraFactory cassandraFactory = new EmbeddedCassandraFactory();
+        Version version = Version.of("4.0-alpha3");
+        Path directory = Paths.get(System.getProperty("user.home") + "/apache-cassandra-4.0-alpha3");
+        cassandraFactory.setArtifact(new DefaultArtifact(version, directory));
+        cassandraFactory.setPort(9142);
```

Comment:
To avoid collision with a real cassandra instead we need to change ports but not only this one. I had a error telling me another app is already run on port xxx.

Reply:
what other port are you talking about ?

Reply:
Just run a classic Cassandra at the same time with unit tests. You will see that 9042 (client protocol port) is not the only one involved. It fails because of collision with other ports.
Cassandra factory allows setting of other ports.